third_party/amd/include/Dialect/TritonAMDGPU/IR/TritonAMDGPUOps.td

```diff
@@ -31,5 +31,64 @@ include "mlir/Dialect/LLVMIR/LLVMOpBase.td"
 include "mlir/Interfaces/InferTypeOpInterface.td"
 include "TritonAMDGPUDialect.td"
 include "TritonAMDGPUAttrDefs.td"
+include "mlir/Interfaces/SideEffectInterfaces.td" // Pure
+include "mlir/Interfaces/ViewLikeInterface.td" // OffsetSizeAndStrideOpInterface
+
+class TritonAMDGPU_Op<string mnemonic, list<Trait> traits = []> :
+    Op<TritonAMDGPU_Dialect, mnemonic, traits>;
+
+def TritonAMDGPU_ViewSliceOp : TritonAMDGPU_Op<"view_slice",
+                                [AttrSizedOperandSegments,
+                                 Pure,
+                                 OffsetSizeAndStrideOpInterface
+                                 ]> {
+  let summary = "view slice operation";
+  let description = [{
+    Represents view of the slice of the tensor in registers. However, unlike 'memdesc_subview' which provides a view in shared memory,
+    'view_slice' specifically slices within registers.
+    Slice of the tensor is required to have the same layout as the original tensor.
+    }];
+
+  let arguments = (ins
+    AnyRankedTensor:$source,
+    Variadic<I32>:$offsets,
```

Comment:
Hmm, do we actually wanting to support dynamic offset/size/stride cases? Isn't this op closer to https://github.com/llvm/llvm-project/blob/6292f117c39b9fc72da4e40328eeeda2aa94a5f2/mlir/include/mlir/Dialect/Vector/IR/VectorOps.td#L1183

Reply:
What I meant is not 2D or 3D; I meant is whether we need to support dynamic offsets/sizes/strides; that is, something like `view_slice %source[%off0, %off1][%size0, %size1]..`. `OffsetSizeAndStrideOpInterface` supports that but I think we only need static offsets/sizes/strides given we are slicing in registers (well tensors in IR)?

@oplavsic wdyt?

Reply:
If we want to make this a more generalized operation we could add support for dynamic offsets/sizes/strides. That would require me to modify the current conversion to llvm and insert llvm instruction to dynamically calculate these. @oplavsic and I talked about this today, and I think for supporting dot-slicing static offsets//sizes/strides are sufficient.

Reply:
Yeah I'm fine with this now. But could you add a validation to error out for non-static cases at least?

Reply:
I could also change the instruction to have them only as attributes like https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/Dialect/Vector/IR/VectorOps.td#L1183

```
%72 = amdgpu.view_slice %arg0 {
  offsets = array<i64: 0,0>,
  sizes = array<i64: 256,16>,
  strides = array<i64: 1,1>
} : tensor<256x128xi32, #blocked1> to tensor<256x16xi32, #blocked1>
```

Is this better or should I keep the old format and verify for non-static?

Reply:
You can keep the old format and verify to reject non-static case for now.